frontend/src/components/Editor/Editor.tsx

```diff
@@ -0,0 +1,25 @@
+import * as monaco from "monaco-editor";
+import {useEffect, useRef} from "react";
+
+function Editor() {
+    function initEditor() {
+        if (ref.current != null && monaco.editor.getEditors().length == 0) {
```

Comment:
here can be TS error "Use before declaration" because the "ref" defined below

Reply:
fixed